File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let product = [];
  for(let i=0;i<nums.length;i++){
    let ans = 1;
    for(let j = 0;j<nums.length;j++){
        if(i == j){
            continue;
        }
        ans = ans * nums[j];
     }
    product.push(ans)
}
return product;
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-0.java
class Solution {
    private int index = 0;


    public List<List<Integer>> subsetsWithDup(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        Arrays.sort(nums);
        backtrack(result, new ArrayList<>(), nums, 0);

        return result;
    }

    public void backtrack(List<List<Integer>> resultSets, List<Integer> tempSet, int[] nums, int start ){
      
        if(resultSets.contains(tempSet)) return;
      
        resultSets.add(new ArrayList<>(tempSet));

        for(int i = start; i < nums.length; i++ ){

            //case of including the number
            tempSet.add(nums[i]);


            //Backtrack the new subset
            backtrack(resultSets, tempSet, nums, i + 1);

           //case of not including the number
           tempSet.remove(tempSet.size() - 1); 

        }   
    }


}